src/BookingConflictConditionFactory.php

```diff
@@ -126,27 +176,30 @@ public function make($config = null)
             );
         }
 
+        $nonBlockingStatuses = $this->_getNonBlockingStatuses();
+        $nonBlockingStatuses = $this->_normalizeArray($nonBlockingStatuses);
+
+        if (count($nonBlockingStatuses) > 0) {
+            // AND the condition to a "booking status NOT IN (...)" sub-condition, where "..." is the list of non-
+            // blocking booking statuses, to make the condition disregard bookings with those statuses.
+            $condition = $b->and(
+                $condition,
+                $b->not(
+                    $b->in(
+                        $b->ef('booking', 'status'),
+                        $b->set($nonBlockingStatuses)
```

Comment:
As far as I've managed to gather, `set` stands for "Sql operator Expression Template". I'm quite confused in the research, and maybe it's faster if I just ask this question directly:

Would the `set` type accept just a list of statuses, in any form, including `Traversable`? this could remove the count check, the array normalization, and would allow the list to be actually traversed only when its contents are needed, i.e. rendered.

Reply:
Yeah unfortunately that's how the `set` expression builder factory works. Probably another cut corner. [See here](https://github.com/RebelCode/rcmod-wp-cqrs/blob/develop/src/Module/WpCqrsModule.php#L276). It can receive variable arguments or a single array argument. Since it detects explicitly arrays only, I have to normalize to array.

Reply:
I anticipated this question so hard that I literally had that answer ready.